plugins/uefi-capsule/efi/generate_sbat.py

```diff
@@ -0,0 +1,145 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2021 Javier Martinez Canillas <javierm@redhat.com>
+# Copyright (C) 2021 Richard Hughes <richard@hughsie.com>
+#
+# SPDX-License-Identifier: LGPL-2.1+
+#
+# pylint: disable=missing-docstring, invalid-name
+
```

Comment:
I think that would be better to split this in two different patches. One that creates an ELF object file with the `.sbat` section and another one that adds the custom linker scripts and uses those instead of `efi.lds` from `gnu-efi`.